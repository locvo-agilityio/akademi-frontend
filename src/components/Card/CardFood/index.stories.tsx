import { StoryObj, Meta } from '@storybook/react';

import CardFood from '.';
import { Box } from '@chakra-ui/react';

const meta = {
  title: 'Components/Cards/CardFood',
  component: CardFood,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => (
    <Box w="312px">
      <CardFood {...args} />
    </Box>
  ),
} satisfies Meta<typeof CardFood>;

export default meta;

type Story = StoryObj<typeof CardFood>;

export const Default: Story = {
  args: {
    name: 'Beef Steak with Fried Potato',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    image: 'https://bit.ly/dan-abramov',
  },
};
