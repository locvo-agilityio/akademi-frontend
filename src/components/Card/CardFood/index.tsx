import { memo } from 'react';
import { Image, Card, CardBody, CardHeader, Text, Box } from '@chakra-ui/react';

interface CardFoodProps {
  name: string;
  content: string;
  image?: string;
}

const CardFood = ({ name, content, image = '' }: CardFoodProps) => (
  <Card
    maxW={312}
    boxShadow="none"
    borderBottom="1px solid"
    borderColor="lightGray"
    borderRadius="none"
  >
    <CardHeader p={0} display="flex" gap={6}>
      <Box w="full" h="160px">
        <Image
          w="full"
          h="full"
          objectFit="cover"
          borderRadius="lg"
          src={image}
          alt="Food Image"
        />
      </Box>
    </CardHeader>

    <CardBody
      p={0}
      display="flex"
      flexDirection="column"
      alignItems="left"
      gap={1}
      mt={6}
    >
      <Text as="p" fontSize="md" fontWeight="bold" color="darkblue">
        {name}
      </Text>
      <Text as="span" color="gray.600" whiteSpace="break-spaces" noOfLines={1}>
        {content}
      </Text>
    </CardBody>
  </Card>
);

export default memo(CardFood);
