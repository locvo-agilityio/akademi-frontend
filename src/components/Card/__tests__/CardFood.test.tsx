import { render } from '@testing-library/react';

// Components
import CardFood from '../CardFood';

describe('CardFood component', () => {
  const mockProps = {
    name: 'Beef Steak with Fried Potato',
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  };

  it('should render correctly with default props', () => {
    const { container } = render(<CardFood {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
