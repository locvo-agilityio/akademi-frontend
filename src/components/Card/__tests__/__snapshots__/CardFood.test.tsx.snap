// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFood component should render correctly with default props 1`] = `
<div>
  <div
    class="chakra-card css-v9162f"
  >
    <div
      class="chakra-card__header css-17udfxa"
    >
      <div
        class="css-6qtbr6"
      >
        <img
          alt="Dan Abramov"
          class="chakra-image css-d7vuui"
        />
      </div>
    </div>
    <div
      class="chakra-card__body css-1i9hxp0"
    >
      <p
        class="chakra-text css-d93k4e"
      >
        Beef Steak with Fried Potato
      </p>
      <span
        class="chakra-text css-1hmu7y7"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </span>
    </div>
  </div>
</div>
`;
